supabase/migrations/20240530291033_add_completed_parcels_view.sql

```diff
@@ -0,0 +1,32 @@
+CREATE OR REPLACE VIEW completed_parcels AS
+WITH completed_events AS (
+    SELECT
+        events.parcel_id,
+        -- If a parcel has multiple successful events, we chose the most recent.
+        MAX(events.timestamp) AS completed_timestamp
+    FROM
+        events
+    WHERE
+        new_parcel_status IN (
+            SELECT
+                event_name
+            FROM
+                status_order
+            WHERE
+                is_successfully_completed_event
+        )
+    -- Ensures each parcel is only counted once.
+    GROUP BY
+        events.parcel_id
```

Comment:
Oh yeah would this not solve the double counting issue with multiple statuses?

Reply:
Yeah it would, that and the MAX line are required together to work. If you group by parcel_id, the things you select now have to be aggregate functions that take in multiple rows and output a single value e.g MAX(timestamp). Just writing timestamp would cause a syntax error. Is that what you mean here?

Reply:
Oh yeah that is how SQL works isn't it 😅 Thanks for explaining, I'm a bit rusty

Reply:
No worries, had no idea about any of those types of sql statements until this ticket, so it's been good to learn